src/nvim/eval.c

```diff
@@ -10779,47 +10779,71 @@ static void f_jobresize(typval_T *argvars, typval_T *rettv)
   rettv->vval.v_number = 1;
 }
 
-// "jobstart()" function
-static void f_jobstart(typval_T *argvars, typval_T *rettv)
+static char **tv_to_argv(typval_T *cmd)
 {
-  rettv->v_type = VAR_NUMBER;
-  rettv->vval.v_number = 0;
-
-  if (check_restricted() || check_secure()) {
-    return;
+  if (cmd->v_type == VAR_STRING) {
+    return shell_build_argv((char *)get_tv_string(cmd), NULL);
   }
 
-  if (argvars[0].v_type != VAR_LIST
-      || (argvars[1].v_type != VAR_DICT && argvars[1].v_type != VAR_UNKNOWN)) {
-    // Wrong argument types
+  if (cmd->v_type != VAR_LIST) {
     EMSG(_(e_invarg));
```

Comment:
I would highly suggest to *never* use `e_invarg`. It is not telling the user what is the error: should be something `Exxx: Expected List or String`.